.github/workflows/python-build-and-upload-packages.yml

```diff
@@ -0,0 +1,63 @@
+name: Python - build and upload packages
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'python-v*'
+  pull_request:
+    # In theory we wouldn't need to create artifacts for each PR, but this makes
+    # it much easier to test this workflow and there are no big cons.
```

Comment:
If the goal was only testing within this PR, then it's also possible to just comment `pull_request` before merging.

Reply:
agreed. updated the comment to reflect this. and will comment that part before merging.